lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp

```diff
@@ -205,7 +205,13 @@ int amf_app_handle_deregistration_req(amf_ue_ngap_id_t ue_id) {
     OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
   }
   // UE context release notification to NGAP
-  amf_app_ue_context_release(ue_context, ue_context->ue_context_rel_cause);
+  if (ue_context->ue_context_rel_cause.ngapCause_u.nas ==
+      ngap_CauseNas_normal_release) {
+    ue_context->ue_context_rel_cause.ngapCause_u.nas = ngap_CauseNas_deregister;
+    amf_app_ue_context_release(ue_context, ue_context->ue_context_rel_cause);
+  } else {
+    amf_app_ue_context_release(ue_context, ue_context->ue_context_rel_cause);
+  }
```

Comment:
```suggestion
  if (ue_context->ue_context_rel_cause.ngapCause_u.nas ==
      ngap_CauseNas_normal_release) {
    ue_context->ue_context_rel_cause.ngapCause_u.nas = ngap_CauseNas_deregister;
  } 
  amf_app_ue_context_release(ue_context, ue_context->ue_context_rel_cause);
  
```

Reply:
This can be refactored

Reply:
Done

Reply:
I do not see any changes here with respect to Alex's comment. How is it `Done`?

Reply:
Fixed, I think these refactor changes were reverted while rebasing latest master.